logic/src/commonMain/kotlin/com/wire/kalium/logic/configuration/E2EISettings.kt

```diff
@@ -23,18 +23,22 @@ import kotlinx.datetime.Instant
 data class E2EISettings(
     val isRequired: Boolean,
     val discoverUrl: String?,
-    val gracePeriodEnd: Instant?
+    val gracePeriodEnd: Instant?,
+    val shouldUseProxy: Boolean,
+    val crlProxy: String?,
 ) {
 
     fun toEntity() = E2EISettingsEntity(
-        isRequired, discoverUrl, gracePeriodEnd?.toEpochMilliseconds()
+        isRequired, discoverUrl, gracePeriodEnd?.toEpochMilliseconds(), shouldUseProxy, crlProxy
     )
 
     companion object {
         fun fromEntity(entity: E2EISettingsEntity) = E2EISettings(
             entity.status,
             entity.discoverUrl,
             entity.gracePeriodEndMs?.let { Instant.fromEpochMilliseconds(it) },
+            entity.shouldUseProxy == true,
```

Comment:
question: is that necessary to compare it? why not returning the value it self?

Reply:
I'd prefer to have as less nullable as possible and in fact this filed should be non-null, but in `Entity` it might be null because of migration from the old versions, that's why i used this ` == true` trick to set `false` as default value even for old-versions.